qt/python/mantidqt/mantidqt/widgets/test/test_fitpropertybrowser.py

```diff
@@ -282,6 +283,21 @@ def test_fit_property_browser_correctly_updates_contents_from_fitted_func(self):
         # it should show a linear line with intercept 2 and gradient 1
         self.assertEqual(str(func.getFunction(0)), 'name=LinearBackground,A0=2,A1=1')
 
+    def test_fit_property_browser_correctly_handles_bin_plots(self):
+        # create & plot workspace
+        fig, canvas, ws = self._create_and_plot_matrix_workspace('ws_name', distribution=True)
+
+        #set bin plot kwargs
+        plot_kwargs = {"axis": MantidAxType.BIN}
+
+        #overplot bin plot
+        plot([ws], wksp_indices=[0], plot_kwargs=plot_kwargs, fig=fig, overplot=True)
+
+        property_browser = self._create_widget(canvas=canvas)
+
+        #if only 1 spectra is returned, bin spectra has been correctly excluded
+        self.assertEqual(len(property_browser._get_allowed_spectra()),1)
```

Comment:
This is handling the message but could you please test the responce for it only a single bin is plotted?

Reply:
Could you elaborate on what you think this test should do - test that the "_get_allowed_spectra()" function returns an empty list if called on a single bin plot?

Reply:
Yes that's what I mean

Reply:
I've added a second assert in the test that checks for this.